src/test/e2e/licensed-users-backfill.test.ts

```diff
@@ -0,0 +1,204 @@
+    await client.query(`CREATE SCHEMA IF NOT EXISTS ${db.schema}`);
+    await client.end();
+}
+
+async function insertEvents(client, events) {
+    const values = events
+        .map(
+            (e) =>
+                `('${e.type}', '${JSON.stringify(e.data || {})}', '${JSON.stringify(
+                    e.pre_data || {},
+                )}', '${e.created_at}', '${e.created_by}')`,
+        )
+        .join(',');
+
+    await client.query(`
+        INSERT INTO events (type, data, pre_data, created_at, created_by)
+        VALUES ${values};
+    `);
+}
+
+describe('licensed_users backfill', () => {
+    jest.setTimeout(15000);
+
+    let client: any;
+    let dbm: any;
+    const config = createTestConfig({
+        db: {
+            ...getDbConfig(),
+            pool: { min: 1, max: 4 },
+            schema: 'licensed_users_test',
+            ssl: false,
+        },
+    });
+
+    beforeAll(async () => {
+        await initSchema(config.db);
+
+        dbm = getInstance(true, {
+            cwd: `${__dirname}/../../`,
+            config: { e2e: { ...config.db, connectionTimeoutMillis: 2000 } },
+            env: 'e2e',
+        });
+
+        await dbm.up('20241114103646-licensed-users.js');
+        client = new Client(config.db);
+        await client.connect();
+        await client.query(`SET search_path = 'licensed_users_test';`);
+    });
+
+    afterAll(async () => {
+        await client.end();
+        await dbm.reset();
+    });
+
+    beforeEach(async () => {
+        await client.query('delete from events;');
+        await client.query('delete from licensed_users;');
+        await client.query(
+            "DELETE FROM migrations WHERE name = '/20241119105837-licensed-users-backfill';",
+        );
+    });
+
+    test('Counts users from their creation date until their deletion date and 30 days', async () => {
+        await insertEvents(client, [
+            {
+                type: 'user-created',
+                data: { email: 'user1@test.com' },
+                created_at: '2024-10-01',
+                created_by: 'test',
+            },
+            {
+                type: 'user-deleted',
+                pre_data: { email: 'user1@test.com' },
+                created_at: '2024-10-05',
+                created_by: 'test',
+            },
+        ]);
+
+        await dbm.up('20241119105837-licensed-users-backfill.js');
+
+        const { rows } = await client.query(
+            "SELECT TO_CHAR(date, 'YYYY-MM-DD') AS date, count FROM licensed_users ORDER BY date;",
+        );
+        expect(rows.find((row) => row.date === '2024-10-01').count).toBe(1);
+        expect(rows.find((row) => row.date === '2024-10-02').count).toBe(1);
+        expect(rows.find((row) => row.date === '2024-10-05').count).toBe(1);
+        expect(rows.find((row) => row.date === '2024-11-04').count).toBe(1);
+        expect(rows.find((row) => row.date === '2024-11-05').count).toBe(0); // 30 days has passed
```

Comment:
This might be the same thing as I pointed out in a separate PR, so feel free to ignore if you don't think it's relevant, but October is a 31-day month. So if you create the user on the 1st, and delete them on the 2nd, the 31st will be the 30th day after deletion. Extrapolating, that means that november 3rd is the 30 day mark. Should we switch this to check November 4th? Or should it be 3rd? Again, because this will in reality differentiate on milliseconds, it probably doesn't matter much. I'm sure youv'e got it covered.

If someone is deleted on the 5th, does the 5th count as the first day of 30 or is the 6th the first day?